rnn_speech_recognition/pytorch/rnnt/decoder.py

```diff
@@ -66,12 +32,34 @@ class RNNTGreedyDecoder(TransducerDecoder):
         cutoff_prob: Skip to next step in search if current highest character
             probability is less than this.
     """
-    def __init__(self, blank_index, model, max_symbols_per_step=30):
-        super().__init__(blank_index, model)
+    def __init__(self, blank_idx, max_symbols_per_step=30, max_symbol_per_sample=None):
+        self.blank_idx = blank_idx
         assert max_symbols_per_step is None or max_symbols_per_step > 0
         self.max_symbols = max_symbols_per_step
+        assert max_symbol_per_sample is None or max_symbol_per_sample > 0
+        self.max_symbol_per_sample = max_symbol_per_sample
+        self._SOS = -1   # start of sequence
+
+    def _pred_step(self, model, label, hidden, device):
+        if label == self._SOS:
+            return model.predict(None, hidden, add_sos=False)
+
+        if label > self.blank_idx:
```

Comment:
Not sure what the intention of this `if` statement and following operation is... it never applies for the reference implementation where `self.blank_idx = num_symbols - 1`. Also, it seems like it could cause erroneous results if `blank_idx` is set to zero.

Reply:
Right. It is left from the old code-base, and it is misleading here. I will remove it.